vignettes/including-static-dashboards.Rmd

```diff
@@ -0,0 +1,103 @@
+
+```
+library(shiny)
+library(crunchy)
+# Render the dashboard
+ui <- fluidPage(
+    includeHTML("dashboard")
+)
+
+server <- function(input, output, session) {}
+
+# Run the application 
+shinyApp(ui = ui, server = server)
+```
+
+### Conditional dashboards
+
+Another great use of Flexdashboards is easily produce different dashboards for different users. To do this we'll make use of Shiny's `renderUI` function to conditionally create a different UI for different types of users.
+
+```
+ui <- fluidPage(
+    uiOutput("dashboard")
+)
+```
+
+This creates the ui object, but most of the work is created on the server side. We can then use the `shinyUser()` function to get a user record for the current viewer of the app. This record includes some useful pieces of information, including the user's name, their email address, and their Crunch preferences. You can use this object to conditionally display different dashboards for different users like so: 
+
+```
+server <- function(input, output, session) {
+    user <- getUser()
+    output$dashboard <- renderUI({
+        if (grepl("acme.com$", user()$email)) {
+            includeHTML("acme_dashboard.html")
+        } else {
+            includeHTML("globex_dashboard.html")
+        }
+    })
+    
+}
+```
+What this does is check the user's email address to identify clients from one company, and then displays a rendered dashboard for that company. If their email isn't a match, they get the Globex dashboard. You can see a full example app in the `/inst` directory. 
+
+
+### Managing Dashboard Renders
+
+The above code assumes that you are going to be re-rendering your Rmd files manually, and uploading the results to the Jupyter server. You can also re-render the Rmd files dynamically as part of the Shiny app by calling `rmarkdown::render()` somewhere in your server function. For example the following app re-renders the dashboard whenever clicks an action button. Note that for this to work you should use the [reactivePoll](https://shiny.rstudio.com/reference/shiny/1.0.2/reactivePoll.html) to ensure that the UI is refreshed after the file is regenerated. 
+
+```
+ui <- sidebarLayout(
+         sidebarPanel(
+             actionButton('refresh', 'Refresh Dashboard'),
+             textOutput("last_render")
+         ),
+    mainPanel(
+             uiOutput("dashboard")
+    )
+)
+
+server <- function(input, output, session) {
+    file <- reactivePoll(1000, session,
+        checkFunc = function() {
+            file.info("acme_dashboard.html")$ctime
+            }, 
+        valueFunc = function() {
+            list(path = "acme_dashboard.html",
+                time = Sys.time())
+            })
+    observeEvent(input$refresh, rmarkdown::render("acme_dashboard.Rmd"))
+    output$last_render <- renderText(paste("Last Render:", file()$time))
+    output$dashboard <- renderUI({
+            includeHTML(file()$path)
+    })
+    
```

Comment:
There's some inconsistent indentation here, and some extra newlines at the bottom.

Reply:
And I see that the same is true in the inst examples.